saleor/graphql/order/mutations/draft_orders.py

```diff
@@ -225,3 +206,118 @@ def mutate(cls, root, info, id):
             type=OrderEvents.PLACED_FROM_DRAFT.value,
             user=info.context.user)
         return DraftOrderComplete(order=order)
+
+
+class DraftOrderLineAdd(BaseMutation):
+    order = graphene.Field(Order, description='A related draft order.')
+    order_line = graphene.Field(OrderLine, description='A newly created order line.')
+
+    class Arguments:
+        id = graphene.ID(
+            required=True,
+            description='ID of the draft order to add the lines to.')
+        input = OrderLineCreateInput(
+            required=True,
+            description="""
+            Variant line input consisting of variant ID and quantity of
+            products.""")
+
+    class Meta:
+        description = 'Add order line to a draft order.'
+
+    @classmethod
+    @permission_required('order.manage_orders')
+    def mutate(cls, root, info, id, input):
+        errors = []
+        order = cls.get_node_or_error(info, id, errors, 'id', Order)
+        variant_id = input['variant_id']
+        variant = cls.get_node_or_error(
+            info, variant_id, errors, 'lines', ProductVariant)
+
+        if not (order or variant):
+            return DraftOrderLineAdd(errors=errors)
+
+        if order.status != OrderStatus.DRAFT:
+            cls.add_error(
+                errors, 'order_id', 'Only draft orders can be edited.')
+
+        line = None
+        if not errors:
+            line = add_variant_to_order(
+                order, variant, input['quantity'], allow_overselling=True)
+            recalculate_order(order)
+        return DraftOrderLineAdd(order=order, order_line=line, errors=errors)
+
+
+class DraftOrderLineDelete(BaseMutation):
+    order = graphene.Field(Order, description='A related draft order.')
+    order_line = graphene.Field(
+        OrderLine, description='An order line that was deleted.')
+
+    class Arguments:
+        id = graphene.ID(
+            description='ID of the order line to delete.', required=True)
+
+    class Meta:
+        description = 'Deletes an order line from a draft order.'
+
+    @classmethod
+    @permission_required('order.manage_orders')
+    def mutate(cls, root, info, id):
+        errors = []
+        line = cls.get_node_or_error(info, id, errors, 'id', OrderLine)
+        if not line:
+            return DraftOrderLineDelete(errors=errors)
+
+        order = line.order
+        if order.status != OrderStatus.DRAFT:
+            cls.add_error(
+                errors, 'id', 'Only draft orders can be edited.')
+        if not errors:
+            variant = line.variant
+            if variant.track_inventory:
+                deallocate_stock(variant, line.quantity)
+            db_id = line.id
+            line.delete()
+            line.id = db_id
+            recalculate_order(order)
+        return DraftOrderLineDelete(
+            errors=errors, order=order, order_line=line)
+
+
+class DraftOrderLineUpdate(ModelMutation):
```

Comment:
I wonder if we shouldn't delete the line, if it's quantity is set to 0?

Reply:
I decided to hold on with implementing this. I think that in terms of API it might be unexpected behavior since there is a dedicated mutation to remove order lines. I'd keep it as it is and see how it works in our frontend.